sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -0,0 +1,185 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+
+/**
+ * @file
+ * @brief <a href="">Reset Manager</a> Device Interface Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Reset Manager generic enable/disable enumeration.
+ *
+ * Enumeration used to enable/disable bits, flags, ...
+ */
+typedef enum dif_rstmgr_enable {
+  kDifRstmgrDisable = 0, /**< disable. */
+  kDifRstmgrEnable,      /**< enable. */
+} dif_rstmgr_enable_t;
+
+/**
+ * Reset Manager peripheral software reset control.
+ */
+typedef enum dif_rstmgr_peripheral_reset {
+  /**
+   * Simple peripheral reset (release straight away).
+   */
+  kDifRstmgrPeripheralReset = 0,
+  kDifRstmgrPeripheralResetHold,    /**< Hold peripheral in reset. */
+  kDifRstmgrPeripheralResetRelease, /**< Release peripheral from reset. */
+} dif_rstmgr_peripheral_reset_t;
+
+/**
+ * Reset Manager all possible reset causes enumeration.
+ *
+ * TODO: no documentation currently available, would be useful
+ *       for writing a more detailed description for the individual causes.
+ */
+typedef enum dif_rstmgr_cause {
```

Comment:
I'm not a fan of using an enum here. What do you think about
```
typedef uint32_t dif_rstmgr_cause_t;
const dif_rstmgr_cause_t kDifRstmgrCausePowerUp = 1;
const dif_rstmgr_cause_t kDifRstmgrCauseLowPowerExit = 2;
...
```

If you want  to stick with enums how about using a utility function to resolve the register offsets rather than building them into the enum values?

Reply:
I don't mind, can go either way, could you however explain why you prefer `const dif_rstmgr_cause_t kDifRstmgrCausePowerUp = 1`? Do you see a potential problem with encoding the values in an enum?

> utility function to resolve the register offsets rather than building them into the enum values?

The enum bit field values in this case only happen to match the offsets in the register definition. In reality there are no guarantees that they will match, hence we do the mapping in the `*_get` function (hopefully I understood your comment correctly).